### DISCUSSION TITLE
Variazione tipologia voti

### DISCUSSION THREAD

**Discussion by Lucaccc** [iisgiua/giuaschool#489] [Richieste di aiuto] [answered]
Buonasera, volevo sapere se fosse possibile variare la tipologia dei voti inseribili...
Mi andrebbe bene anche modificare soltanto il testo che appare nel momento in cui si visualizza un voto; c'è un modo per poterlo fare senza danneggiare il registro?
Grazie.

**Comment by trinko** [comment]
Non mi è molto chiaro cosa intendi fare. 
Comunque, in generale puoi modificare la visualizzazione andando a modificare il _template_ della pagina. Se intendi procedere in questo modo, devi anzi tutto chiarire quali pagine devono essere modificate: voti mostrati a genitore/alunno, pagina inserimento voti docente, pagina medie del coordinatore...
Per le personalizzazione dei _template_ vedi: [Personalizzazione delle pagine del registro](https://iisgiua.github.io/giuaschool-docs/customize.html#personalizzazione-delle-pagine-del-registro)

**Reply by Lucaccc** [reply]
Vorrei modificare la dicitura che compare nella sezione "Tipologia" del voto, senza però modificare le altre funzioni del registro.
Mi spiego meglio: senza andare a modificare pulsanti etc dei voti scritto, orale e pratico, vorrei che, ad esempio, al posto della dicitura "scritto" corrispondesse "Valutazione Sommativa", ma solo quando un voto deve essere visualizzato.
Ciò nasce dal fatto che nella mia Scuola non sono in uso le tipologie S/O/P, ma Valutazione Sommativa e Formativa e mi rendo conto che nel registro non possono essere modificate queste tipologie al posto di quelle standard; per questo motivo mi sarebbe sufficiente modificarle la dicitura che compare nel momento della visualizzazione del voto.
Quali template devo andare a modificare?
Aggiornamento: sono risucito a farlo per lato genitore/alunni, ma per i docenti non trovo quale template devo modificare.

**Reply by trinko** [reply, marked as answer]
Il template principale è **templates/lezioni/voti_quadro.html.twig** 

Ti raccomando di copiare i template modificati in **PERSONAL/templates/**: in questo modo, utilizzando l'aggiornamento automatizzato non c'è pericolo che vengano sovrascritti.

**Comment by trinko** [comment]
Dovresti poi modificare anche il file delle traduzioni delle etichette, utilizzate in varie parti del codice. 
Il file da modificare è **translations/messages+intl-icu.it.yml** e le etichette sono le seguenti:
```
label.voti_O: Orale
label.voti_P: Pratico
label.voti_S: Scritto/Grafico
```
